src/record/record-transition.ts

```diff
@@ -368,15 +368,22 @@ export default class RecordTransition {
     if (error) {
       this.onFatalError(error)
     } else if (this.isDestroyed === false) {
+      // // Get values that should not be broadcast
+      const { isWriteAck, correlationId } = this.currentStep.message
+
+      // // Delete values that should not be broadcast
       this.currentStep.message.isWriteAck = false
-      delete this.currentStep.message.correlationId
+      this.currentStep.message.correlationId = undefined
       this.recordHandler.broadcastUpdate(
```

Comment:
We could do

`{ ...mesage, isWriteAck: false, correlationId: undefined }`

I think the cost of assigning a property vs the cost of object creation isn't much different anymore. Thoughts?

Reply:
I think it would be wisest to write it in the most clear and concise way possible, and then I think doing what you suggest would be good. Fixing it!

Reply:
I fixed it, but a few tests expected `correlationId` to not exist (`{}` vs. `{correlationId: undefined}`. So in the solution I unpack the original message and specifically delete `correlationId`.